sdk/nodejs/runtime/rpc.ts

```diff
@@ -275,13 +277,17 @@ export async function serializeProperty(ctx: string, prop: Input<any>, dependent
         return isKnown ? value : unknownValue;
     }
 
+    if (Resource.isInstance(prop)) {
+        // Keep track of this resource as a dependency.
+        dependentResources.add(prop);
```

Comment:
we want to do this regardless of whether this is a custom or component resource.